docker/policyconfiguration/naming.go

```diff
@@ -43,14 +45,25 @@ func DockerReferenceNamespaces(ref reference.Named) []string {
 	// iteration matches the host name (for any namespace).
 	res := []string{}
 	name := ref.Name()
+	var subdomain bool
 	for {
-		res = append(res, name)
-
-		lastSlash := strings.LastIndex(name, "/")
-		if lastSlash == -1 {
-			break
+		if !subdomain {
+			res = append(res, name)
+			lastSlash := strings.LastIndex(name, "/")
+			if lastSlash == -1 {
+				s := "true"
+				subdomain, _ = strconv.ParseBool(s)
+				continue
+			}
+			name = name[:lastSlash]
+		} else {
+			res = append(res, name)
+			firstDot := strings.Index(name, ".")
+			if firstDot == -1 {
+				break
+			}
+			name = fmt.Sprint("*", name[firstDot-1:])
```

Comment:
Given a `foo.bar`, this turns `name` into `*.bar`, i.e. the same number of dots, so the loop never terminates AFAICS. The tests running out of memory are consistent with that.